core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPruneOrderByInWindowAggregation.java

```diff
@@ -0,0 +1,103 @@
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.ResolvedFunction;
+import io.trino.spi.connector.SortOrder;
+import io.trino.sql.ir.Reference;
+import io.trino.sql.planner.OrderingScheme;
+import io.trino.sql.planner.Symbol;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.planner.plan.DataOrganizationSpecification;
+import io.trino.sql.planner.plan.WindowNode;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.metadata.TestMetadataManager.createTestMetadataManager;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.sort;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.specification;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.window;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.windowFunction;
+import static io.trino.sql.planner.plan.WindowNode.Frame.DEFAULT_FRAME;
+import static io.trino.sql.tree.SortItem.NullOrdering.LAST;
+import static io.trino.sql.tree.SortItem.Ordering.ASCENDING;
+
+public class TestPruneOrderByInWindowAggregation
+        extends BaseRuleTest
+{
+    private static final Metadata METADATA = createTestMetadataManager();
+
+    @Test
+    public void testBasics()
+    {
+        tester().assertThat(new PruneOrderByInWindowAggregation(METADATA))
+                .on(this::buildWindowNode)
+                .matches(
+                        window(
+                                windowMatcherBuilder -> windowMatcherBuilder
+                                        .specification(specification(
+                                                ImmutableList.of("key"),
+                                                ImmutableList.of(),
+                                                ImmutableMap.of()))
+                                        .addFunction(
+                                                "avg",
+                                                windowFunction("avg", ImmutableList.of("input"), DEFAULT_FRAME))
+                                        .addFunction(
+                                                "array_agg",
+                                                windowFunction("array_agg", ImmutableList.of("input"), DEFAULT_FRAME, ImmutableList.of(sort("input", ASCENDING, LAST)))),
+                                values("input", "key", "keyHash", "mask")));
+    }
+
+    private WindowNode buildWindowNode(PlanBuilder planBuilder)
```

Comment:
Consider inlining this. That way, it would be more consistent with the tests code style.